arch/endian_test.go

```diff
@@ -0,0 +1,60 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package arch_test
+
+import (
+	"encoding/binary"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/arch"
+)
+
+type endianTestSuite struct{}
+
+var _ = Suite(&endianTestSuite{})
+
+func (s *endianTestSuite) TestEndian(c *C) {
+	for _, t := range []struct {
+		arch   string
+		endian binary.ByteOrder
+	}{
+		{"ppc", binary.BigEndian},
+		{"ppc64", binary.BigEndian},
+		{"s390x", binary.BigEndian},
+		{"i386", binary.LittleEndian},
+		{"amd64", binary.LittleEndian},
+		{"arm", binary.LittleEndian},
+		{"arm64", binary.LittleEndian},
+		{"riscv64", binary.LittleEndian},
+	} {
+		restore := arch.MockRuntimeGOARCH(t.arch)
+		defer restore()
+
+		c.Check(arch.Endian(), Equals, t.endian)
+	}
+}
+
+func (s *endianTestSuite) TestEndianErrors(c *C) {
+	restore := arch.MockRuntimeGOARCH("unknown-arch")
+	defer restore()
+
+	c.Check(func() { arch.Endian() }, Panics, "unknown architecture amd64")
```

Comment:
I think this should be:

```suggestion
	c.Check(func() { arch.Endian() }, Panics, "unknown architecture unknown-arch")
```